app/controllers/admin/export/document_controller.rb

```diff
@@ -21,9 +21,21 @@ def edition_associations(edition)
              }
     output[:associations] = {}
     associations = edition.class.reflect_on_all_associations.map(&:name)
+
     associations.each do |association|
-      output[:associations][association] = edition.public_send(association)
+      if association == :images
+        edition_images = edition.public_send(association)
+        output[:associations][association] = complete_images_hash(edition_images)
+      else
+        output[:associations][association] = edition.public_send(association)
+      end
     end
     output
   end
+
+  def complete_images_hash(edition_images)
+    edition_images.map do |image|
+      image.attributes.merge("file_url" => image.image_data.file_url)
```

Comment:
Btw you can call [`.url`] on an image object to avoid the need to access image_data: https://github.com/alphagov/whitehall/blob/2f580a2ded3964394d5de9d08a28cda12fecd9d2/app/models/image.rb#L14-L16

Also It seems a bit weird to be merging into the attributes of a model here. It might be better to use as_json to convert it into a hash. So this could be `edition_images.map { |i| i.as_json(methods: :url) }` and may mean you avoid the need for an extra method.

Reply:
Ah ah yep that is better & I didn't know `as_json` could do such things... done!